app/services/reference_generator.rb

```diff
@@ -14,7 +14,7 @@ def attributes
 
   def business_entity
     @business_entity ||= BusinessEntity.where(
```

Comment:
I think here we'd ideally have some method on `BusinessEntity` to give us a valid one for a particular `office_id` and `jurisdiction_id`, so that we don't have to repeat those queries outside `BusinessEntity`. The valid dates are internal thing to this object and are not concern of most parts of the app.